### PR TITLE
implement mint stable credit

### PR DESCRIPTION
**Pull request by athiwatp**

### CODE REVIEW

File: contracts/modules/core/DigitalReserveSystem.sol [hunk 1]
```diff
@@ -125,6 +125,49 @@ contract DigitalReserveSystem is IDRS {
         return true;
     }
 
+    function mintStableCredit(
+        uint256 stableCreditAmount,
+        string calldata assetCode
+    ) external payable returns (bool) {
+        require(heart.isTrustedPartner(msg.sender), "only trusted partner can mint the stable credit");
+
+        StableCredit stableCredit = heart.getStableCreditById(getStableCreditId(assetCode));
```
Comment (jura-b) [inline contracts/modules/core/DigitalReserveSystem.sol]: StableCredit stableCredit = heart.getStableCreditBId(Hasher.stableCreditId(assetCode)
)

File: contracts/modules/core/DigitalReserveSystem.sol [hunk 2]
```diff
@@ -125,6 +125,49 @@ contract DigitalReserveSystem is IDRS {
         return true;
     }
 
+    function mintStableCredit(
+        uint256 stableCreditAmount,
+        string calldata assetCode
+    ) external payable returns (bool) {
+        require(heart.isTrustedPartner(msg.sender), "only trusted partner can mint the stable credit");
```
Comment (jura-b) [inline contracts/modules/core/DigitalReserveSystem.sol]: use modifier

File: contracts/modules/core/DigitalReserveSystem.sol [hunk 3]
```diff
@@ -125,6 +125,49 @@ contract DigitalReserveSystem is IDRS {
...
+
+        StableCredit stableCredit = heart.getStableCreditById(getStableCreditId(assetCode));
+        require(address(stableCredit) != address(0), "stableCredit not exist");
```
Comment (jura-b) [inline contracts/modules/core/DigitalReserveSystem.sol]: add prefix DigitalReserveSystem.mintStableCredit

File: contracts/modules/core/DigitalReserveSystem.sol [hunk 4]
```diff
@@ -125,6 +125,49 @@ contract DigitalReserveSystem is IDRS {
...
+
+        bytes32 collateralAssetCode = stableCredit.collateralAssetCode();
+        bytes32 linkId = keccak256(abi.encodePacked(collateralAssetCode, stableCredit.peggedCurrency()));
```
Comment (jura-b) [inline contracts/modules/core/DigitalReserveSystem.sol]: use Hasher

File: contracts/modules/core/DigitalReserveSystem.sol [hunk 5]
```diff
@@ -125,6 +125,49 @@ contract DigitalReserveSystem is IDRS {
...
+
+        require(heart.getCollateralAsset(collateralAssetCode) == stableCredit.collateral(), "collateralAsset must be the same");
+        require(heart.getPriceFeeders().getMedianPrice(linkId) != 0, "median price ref mut not be zero");
```
Comment (jura-b) [inline contracts/modules/core/DigitalReserveSystem.sol]: price of link must have value more than 0

File: contracts/modules/core/Heart.sol [hunk 6]
```diff
@@ -63,12 +63,12 @@ contract Heart is IHeart {
 
 
     modifier onlyGovernor() {
-        require(isGovernor(msg.sender), "WhitelistGovernorRole: caller does not have the Whitelist governor role");
+        require(isGovernor(msg.sender), "The message sender is not found or does not have sufficient permission");
```
Comment (jura-b) [inline contracts/modules/core/Heart.sol]: prefix Heart.onlyGovernor:

File: contracts/modules/core/Heart.sol [hunk 7]
```diff
@@ -63,12 +63,12 @@ contract Heart is IHeart {
...
         _;
     }
 
     modifier onlyTrustedPartner() {
-        require(isTrustedPartner(msg.sender), "WhitelistTrustedPartnerRole: caller does not have the Whitelist trusted partner role");
+        require(isTrustedPartner(msg.sender), "The message sender is not found or does not have sufficient permission");
```
Comment (jura-b) [inline contracts/modules/core/Heart.sol]: prefix Heart.onlyGovernor:

**Review by Kafakk — APPROVED** [review]

**Pull request merged**